File: src/main/java/com/lcaohoanq/nocket/domain/token/Token.java
package com.lcaohoanq.nocket.domain.token;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.lcaohoanq.nocket.domain.user.User;
import com.lcaohoanq.nocket.base.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
@Entity
@Table(name = "tokens")
public class Token extends BaseEntity {
    
    @Column(name = "token", length = 255)
    private String token;

    @Column(name = "refresh_token", length = 255)
    private String refreshToken;

    @Column(name = "token_type", length = 50)
    private String tokenType;

    @Column(name = "expiration_date")
    private LocalDateTime expirationDate;

    @Column(name = "refresh_expiration_date")
    private LocalDateTime refreshExpirationDate;

    @Column(name = "is_mobile")
    private boolean isMobile;

    private boolean revoked;
    private boolean expired;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

}

File: src/main/kotlin/com/lcaohoanq/nocket/domain/token/Token.kt
package com.lcaohoanq.nocket.domain.token

import com.lcaohoanq.nocket.base.entity.BaseEntity
import com.lcaohoanq.nocket.domain.user.User
import jakarta.persistence.*
import org.hibernate.envers.Audited
import java.time.LocalDateTime

@Audited
@Entity
@Table(name = "tokens")
class Token : BaseEntity() {

    @Column(name = "token", length = 255)
    var token: String? = null

    @Column(name = "refresh_token", length = 255)
    var refreshToken: String? = null

    @Column(name = "token_type", length = 50)
    var tokenType: String? = null

    @Column(name = "expiration_date")
    var expirationDate: LocalDateTime? = null

    @Column(name = "refresh_expiration_date")
    var refreshExpirationDate: LocalDateTime? = null

    @Column(name = "is_mobile")
    var isMobile: Boolean = false
    var revoked: Boolean? = false
    var expired = false

    @ManyToOne
    @JoinColumn(name = "user_id")
    var user: User? = null


}